config/initializers/houston_async.rb

```diff
@@ -4,7 +4,8 @@ module Houston
   def self.async(do_async=true)
     return yield unless do_async
     Thread.new do
-      ActiveRecord::Base.connection_pool.with_connection do
+      ActiveRecord::Base.connection_pool.with_connection do |connection|
+        connection.verify!
```

Comment:
> Checks whether the connection to the database is still active (i.e. not stale). This is done under the hood by calling #active?. If the connection is no longer active, then this method will reconnect to the database.

Good call 👍. I was thinking the `!` method would just throw a different type of exception 😄